package/MDAnalysis/lib/log.py

```diff
@@ -210,6 +210,93 @@ def _guess_string_format(template):
         return _legacy_format
 
 
+class _ProgressBar(object):
+    def __init__(self, pm):
+        self._pm = pm
+        self.default_length = 20
+
+    def __format__(self, format_spec=None):
+        split_spec = format_spec.split('-')
+        if len(split_spec) == 2:
+            full_symbol, empty_symbol = split_spec[0]
+            length = int(split_spec[1])
+        elif len(split_spec) == 1:
+            try:
+                length = int(format_spec)
+            except ValueError:
+                length = self.default_length
+            full_symbol = '#'
+            empty_symbol = '='
+        else:
+            raise ValueError()
+
+        n_full = int(length * self._pm.percentage / 100)
+        n_empty = int(length - n_full)
+        return full_symbol * n_full + empty_symbol * n_empty
+
+
+class _TimeDelta(datetime.timedelta):
+    _fmt_map = {
+        'h': 'remain_hours',
+        'm': 'remain_minutes',
+        's': 'remain_seconds',
+        'D': 'in_days',
+        'H': 'in_hours',
+        'M': 'in_minutes',
+        'S': 'in_seconds',
+    }
+
+    def _sub_format(self, format_spec, i):
+        if format_spec[i + 1] == '%':
+            return i + 2, '%'
+        key = format_spec[i + 1]
+        i += 2
+        return i, str(getattr(self, self._fmt_map[key]))
+
+    def __format__(self, format_spec):
+        if not format_spec:
+            return self.__str__()
+
+        formatted = ''
+        i = 0
+        while i < len(format_spec):
+            if format_spec[i] != '%':
+                formatted += format_spec[i]
+                i += 1
+            else:
+                i, chunck = self._sub_format(format_spec, i)
+                formatted += chunck
+        return formatted
+
+    @property
+    def in_days(self):
```

Comment:
I just killed a script that ran for two weeks. Sometime a naive implementation you are confident about is more practical that spending the same amount of time trying to optimize the code.